src/main/scala/workflow/graph/PipelineDatasetOut.scala

```diff
@@ -0,0 +1,25 @@
+package workflow.graph
+
+import org.apache.spark.rdd.RDD
+
+/**
+ * This class is a lazy wrapper around the output of a pipeline that was passed an RDD as input.
+ *
+ * Under the hood, it extends [[GraphExecution]] and keeps track of the necessary execution plan.
+ */
+class PipelineDatasetOut[T] private[graph] (executor: GraphExecutor, sink: SinkId, source: Option[(SourceId, RDD[_])])
```

Comment:
I'm open to suggestions about better names! And yes, the constructor is being marked `private[graph]`.